src/main/java/com/espro/flink/consul/jobregistry/ConsulRunningJobsRegistry.java

```diff
@@ -33,32 +38,50 @@ public ConsulRunningJobsRegistry(Supplier<ConsulClient> client, ConsulSessionHol
 	}
 
 	@Override
-	public void setJobRunning(JobID jobID) throws IOException {
-		storeJobStatus(jobID, JobSchedulingStatus.RUNNING);
+	public void createDirtyResult(JobResultEntry jobResultEntry) throws IllegalStateException {
+		storeJobStatus(jobResultEntry.getJobId(), JobStatus.DIRTY.getValue());
 	}
 
 	@Override
-	public void setJobFinished(JobID jobID) throws IOException {
-		storeJobStatus(jobID, JobSchedulingStatus.DONE);
+	public void markResultAsClean(JobID jobID) throws IOException, NoSuchElementException {
+		storeJobStatus(jobID, JobStatus.CLEAN.getValue());
 	}
 
 	@Override
-	public JobSchedulingStatus getJobSchedulingStatus(JobID jobID) throws IOException {
-        GetValue value = client.get().getKVValue(path(jobID)).getValue();
-		return value == null ? JobSchedulingStatus.PENDING : JobSchedulingStatus.valueOf(value.getDecodedValue());
+	public boolean hasDirtyJobResultEntry(JobID jobID) throws IOException {
+		GetValue value = client.get().getKVValue(path(jobID)).getValue();
+		if (value == null) {
+			return true;
+		}
+		return value.getDecodedValue().equals(JobStatus.DIRTY.getValue());
 	}
 
 	@Override
-	public void clearJob(JobID jobID) throws IOException {
-        client.get().deleteKVValue(path(jobID));
+	public boolean hasCleanJobResultEntry(JobID jobID) throws IOException {
+		GetValue value = client.get().getKVValue(path(jobID)).getValue();
+		if (value == null) {
+			return true;
+		}
+		return value.getDecodedValue().equals(JobStatus.CLEAN.getValue());
+	}
+
+	@Override
+	public Set<JobResult> getDirtyResults() throws IOException {
+		return null;
+	}
+
+	public void markResultAsDirty(JobID jobID) {
```

Comment:
Why is this method added here, because it seems that it is only used by test classes

Reply:
Yes, it's for testing purpose.

Reply:
done